### PR TITLE
Added clang-tidy GitHub action

### PR DESCRIPTION
**Pull request by zjeffer**

### CODE REVIEW

**Comment by zjeffer** [comment]
@Alexays can you update the `debian` Docker image now that https://github.com/Alexays/Waybar/pull/2742 is merged? Thanks

**Comment by Alexays** [comment]
> @Alexays can you update the `debian` Docker image now that #2742 is merged? Thanks

Done! :)

**Comment by zjeffer** [comment]
Thanks! I'm still fixing some small issues with the action, I'll ping you once it's ready to merge.

**Comment by zjeffer** [comment]
@Alexays I included a change to a cpp file to give an example of a clang-tidy warning:

image

Let me know if this is ok, and I can then revert that change.

Right now the action is still using my own fork of the cpp-linter-action, we should hold off on merging until [this PR](https://github.com/cpp-linter/cpp-linter/pull/44) is merged and a new cpp-linter-action release is made.

**Comment by Alexays** [comment]
LGTM!
Nice :)

File: .github/workflows/clang-tidy.yml [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+    runs-on: ubuntu-latest
+    container:
+      image: alexays/waybar:debian
+    steps:
+      - uses: actions/checkout@v3
+      - name: configure
+        run: |
+          meson -Dcpp_std=c++20 build  # necessary to generate compile_commands.json
+          ninja -C build               # necessary to find certain .h files (xdg, wayland, etc.)
+      - uses: zjeffer/cpp-linter-action@test/process-compilation-database
```
Comment (2bndy5) [inline .github/workflows/clang-tidy.yml:16]: The dev branch that this fork uses no longer exists. The fixes it introduced are now part of the v2.7.5 release; please update this line:
```suggestion
      - uses: cpp-linter/cpp-linter-action@v2
```

**Pull request merged**